src/iarc7_sensors/obstacle_detector.py

```diff
@@ -0,0 +1,358 @@
+#!/usr/bin/env python2
+
+from iarc7_msgs.msg import (Obstacle,
+                            ObstacleArray)
+from geometry_msgs.msg import Point
+from nav_msgs.msg import Odometry
+from sensor_msgs.msg import LaserScan
+from visualization_msgs.msg import Marker, MarkerArray
+
+import numpy as np
+np.set_printoptions(threshold='nan')
+
+from collections import deque
+import numpy.linalg as linalg
+import rospy
+import scipy.interpolate
+import scipy.optimize
+from tf import transformations
+import tf2_ros
+import uuid
+
+def slerp(q0, q1, t):
+    q0 = q0 / linalg.norm(q0)
+    q1 = q1 / linalg.norm(q1)
+
+    dot = q0.dot(q1)
+
+    if abs(dot) > 0.9995:
+        result = q0 + np.outer(t, q1 - q0)
+        return result / np.expand_dims(linalg.norm(result, axis=1), axis=-1)
+
+    if dot < 0:
+        q1 *= -1
+        dot *= -1
+
+    dot = min(1, max(dot, -1))
+    theta = t * np.arccos(dot)
+    q2 = q1 - q0*dot
+    q2 /= linalg.norm(q2)
+    return np.outer(np.cos(theta), q0) + np.outer(np.sin(theta), q2)
+
+def scan_callback(msg):
+    if not msg_queue or msg_queue[-1].header.stamp < msg.header.stamp:
+        msg_queue.append(msg)
+    else:
+        rospy.logerr('Rejecting scan with old timestamp: %s'%msg.header.stamp)
+
+def get_circle(points, radius, drone_pos, settings):
+    assert points.size
+
+    mean_point = np.mean(points, axis=0)
+    guess_pos_offset = mean_point - drone_pos
+    guess_pos_offset *= (
+            (linalg.norm(guess_pos_offset) + settings['obst_radius'])
```

Comment:
Does this do what its supposed to do? Because it simplifies to guess_pos_offset * (1 + obst_radius/||guess_pose_offset||). The current version calculates the norm twice instead of once.

Reply:
Yep, it's taking a vector to the boundary of the obstacle and lengthening it into a vector to the center of the obstacle.  I think it's right